engine/vcs/vcs_test.go

```diff
@@ -184,6 +184,14 @@ func TestRepoRootForImportPath(t *testing.T) {
 				Root: "gitlab.com/testguigui1/dagger-public-sub/mywork.git",
 			},
 		},
+		{
+			"gitlab.com/dagger-modules/test/more/dagger-test-modules-public/../../..",
+			&RepoRoot{
+				VCS:  vcsGit,
+				Repo: "https://gitlab.com/dagger-modules/test/more/dagger-test-modules-public",
+				Root: "gitlab.com/dagger-modules/test/more/dagger-test-modules-public",
+			},
+		},
```

Comment:
What about something like `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/dagger/weirdpath/..`

This *should* resolve the `Root` to `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/dagger`, but I think by stripping the `../` early, we're going to get the wrong result, and instead get `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/weirdpath` which doesn't exist.

Should we re-append the stripped `../`s at the end to the `Root`? I think that might resolve this.

Reply:
So, not sure to get this one.

We do strip, but we strip in `RepoRootForImportDynamic`, which is right before our call to the ref + `go-get=1`.
That way, we are sure that the endpoint does not normalize it suddenly.

It does actually work, and I _believe_ that this is what we actually want.
_Example_: 

This works: `https://gitlab.com/dagger-modules/test/more/dagger-test-modules-public/weirdpath?go-get=1`, but, this `https://gitlab.com/dagger-modules/test/more/dagger-test-modules-public/weirdpath/../../?go-get=1`  does not.

_Working example:`gitlab.com/dagger-modules/test/more/dagger-test-modules-public/weirdpath?go-get=1` _
<img width="1187" alt="image" src="https://github.com/user-attachments/assets/9b061541-b02e-468a-b58e-3fe7a9d25904">

_Failure: `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/weirdpath/../../?go-get=1`_
<img width="1158" alt="image" src="https://github.com/user-attachments/assets/e3ea1b49-bb83-41d4-bfe5-ee1690f6d0d3">


Why ? Because it shortens (at normalization) the ref, making the ref outside of the real repo:

The root is: 
- `gitlab.com/dagger-modules/test/more/dagger-test-modules-public` ; `go-get=1` returns the proper Root
- `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/weirdpath/depth1/depth2...` : `go-get=1` returns the proper Root (any subpath inside a real public repo)
- `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/weirpath/..`: would work too, as `go-get=1` is still inside the repo scope, even after normlization into `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/?go-get=1`

Now, without trimming:
- `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/..` would lead to `gitlab.com/dagger-modules/test/more?go-get=1`, which does not fail but return the wrong root, as shown on the failure picture

But with trimming: `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/..` becomes `gitlab.com/dagger-modules/test/more/dagger-test-modules-public/` which leads to this dynamic query: `gitlab.com/dagger-modules/test/more/dagger-test-modules-public?go-get=1` which is correct

Reply:
Yup, you're entirely right. This looks good to me!